### PR TITLE
Fix CProver.nondetWithoutNull not behaving as expected

### PR DESCRIPTION
**Pull request by romainbrenguier**
The way this call was replaced in JBMC does not behave as expected,
i.e. the results could be null.
Not replacing it and loading the CProver class in
src/java_bytecode/library/src/org/cprover/CProver.java gives what is
actually expected.

### CODE REVIEW

File: regression/jbmc-cover/nondetWithoutNull/Test.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// This file needs to be compiled with
+// src/java_bytecode/library/src/org/cprover/CProver.java
+import org.cprover.CProver;
+
+public class Test {
+    public static int check() {
+	Object o = CProver.nondetWithoutNull();
+
+	if(o == null)
```
Comment (thk123) [inline regression/jbmc-cover/nondetWithoutNull/Test.java]: Formatting seems out

Comment (thk123) [reply regression/jbmc-cover/nondetWithoutNull/Test.java]: Why not use `assert(o!=null)` and then check the verification successful? I think this is simpler and less error prone than relying on specific line numbers

**Review by thk123 — APPROVED** [review]
I wonder if it would make more sense to have the non-det with null implemented in java since currently we do something like:

```
if(nondet)
  a = null
else 
  a = new A();
assume(a != null)
```

**Comment by peterschrammel** [comment]
@romainbrenguier, what's the status of this?

**Comment by romainbrenguier** [comment]
> @romainbrenguier, what's the status of this?

This is outdated, and we have workarounds which work fine now. Since this PR is not a clean fix I'm closing it.

**Pull request closed without merging**